src/FSharp.Control.TaskSeq/TaskSeq.fsi

```diff
@@ -154,414 +175,872 @@ module TaskSeq =
+    /// <summary>
+    /// Builds an <see cref="array" /> from the input task sequence in <paramref name="source" />.
+    /// This function is blocking until the sequence is exhausted and will then properly dispose of the resources.
+    /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting array.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val toArray: source: taskSeq<'T> -> 'T[]
 
     /// <summary>
-    /// Returns the task sequence <paramref name="source" /> as an F# <see cref="seq" />, that is, an
-    /// <see cref="IEnumerable&lt;'T>" />. This function is blocking at each <see cref="yield" />, but otherwise
-    /// acts as a normal delay-executed sequence.
-    /// It will then dispose of the resources.
+    /// Views the task sequence in <paramref name="source" /> as an F# <see cref="seq" />, that is, an
+    /// <see cref="IEnumerable&lt;'T>" />. This function is blocking at each <see cref="yield" /> or call
+    /// to <see cref="IEnumerable&lt;'T>/MoveNext()" /> in the resulting sequence.
+    /// Resources are disposed when the sequence is disposed, or the sequence is exhausted.
     /// </summary>
     ///
     /// <param name="source">The input task sequence.</param>
     /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val toSeq: source: taskSeq<'T> -> seq<'T>
 
-    /// Unwraps the taskSeq as a Task<array<_>>. This function is non-blocking.
+    /// <summary>
+    /// Builds an <see cref="array" /> asynchronously from the input task sequence in <paramref name="source" />.
+    /// This function is non-blocking while it builds the array.
+    /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting array.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val toArrayAsync: source: taskSeq<'T> -> Task<'T[]>
 
-    /// Unwraps the taskSeq as a Task<list<_>>. This function is non-blocking.
+    /// <summary>
+    /// Builds an F# <see cref="list" /> asynchronously from the input task sequence in <paramref name="source" />.
+    /// This function is non-blocking while it builds the list.
+    /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting list.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val toListAsync: source: taskSeq<'T> -> Task<'T list>
 
-    /// Unwraps the taskSeq as a Task<ResizeArray<_>>. This function is non-blocking.
+    /// <summary>
+    /// Builds a resizable array asynchronously from the input task sequence in <paramref name="source" />.
```

Comment:
I think this should have a remark conveying that `toArray`/`toList` should normally be used as returning a mutable buffer with potential excess capacity is rarely a good default (yes array is mutable too, but not really!) is not a normal good thing
```suggestion
    /// Gathers items into a ResizeArray (`System.Collections.Generic.List`) asynchronously from the input task sequence in <paramref name="source" />.
```

Reply:
Not sure about your comment here. Using `toArray/toList` are certainly preferred, but that's the same as with `seq`.

Reply:
Tx. Improved in #136.